### PR TITLE
NOISSUE - Rename clients to things

### PR DESCRIPTION
**Pull request by mijicd**
`Client` has been renamed to `Thing` as it is a better name for the concept. All related references including the service itself have been renamed.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -5,32 +5,29 @@ Mainflux IoT platform is comprised of the following services:
...
+connecting them.
+
+`Thing` represents devices and applications connected to Mainflux. There are
+two types of "things" supported at the moment: `device` and `app`. While device
+is used to represent any physical device connected to the platform, app represents
+any 3rd party service that uses the platform for message exchange with other
+"things". 
 
 `Channel` represents a communication channel. It serves as message topic that
 can be consumed by all of the clients connected to it.
```
Comment (nmarcetic) [inline docs/architecture.md]: rename to things.

File: ws/README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ services:
     ports:
       - [host machine port]:[configured port]
     environment:
-      MF_CLIENTS_URL: [Clients service URL]
+      MF_THINGS_URL: [Clients service URL]
```
Comment (nmarcetic) [inline ws/README.md]: Rename to things service URL

File: ws/README.md [hunk 3]
```diff
@@ -48,10 +48,10 @@ make ws
 make install
 
 # set the environment variables and run the service
-MF_CLIENTS_URL=[Clients service URL] MF_NATS_URL=[NATS instance URL] MF_WS_ADAPTER_PORT=[Service WS port] $GOBIN/mainflux-ws
+MF_THINGS_URL=[Clients service URL] MF_NATS_URL=[NATS instance URL] MF_WS_ADAPTER_PORT=[Service WS port] $GOBIN/mainflux-ws
```
Comment (nmarcetic) [inline ws/README.md]: Rename to Things service URL

**Review by nmarcetic — CHANGES_REQUESTED** [review]

**Review by nmarcetic — APPROVED** [review]

**Comment by drasko** [comment]
@mijicd can you please resolve the conflicts?

**Pull request merged**